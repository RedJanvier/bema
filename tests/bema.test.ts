import './__setup'
import { noop } from 'lodash'
import { stripKeys } from './__helpers'

test('can define a case', async () => {
  bema.group('a').case().run(f1)
  await run(bema)
  expect(f1.mock.calls.length).toBeGreaterThan(0)
})

test('can define cases in groups', async () => {
  const bema2 = bema.parameter('name')
  const ga = bema2.group('a')
  ga.case({ name: '1' }).run(noop)
  ga.case({ name: '2' }).run(noop)
  const gb = bema2.group('b')
  gb.case({ name: '3' }).run(noop)
  gb.case({ name: '4' }).run(noop)
  const report = await run(bema)
  expect(stripKeys(['stats', 'time'], report.groups)).toMatchSnapshot()
})

test('afterEach is called after each case', async () => {
  let ga = bema.afterEach(f1).parameter('name').group('a')
  ga.case({ name: '1' }).run(noop)
  ga.case({ name: '2' }).run(noop)
  ga.case({ name: '3' }).run(noop)
  await run(bema)
  expect(f1.mock.calls.length).toEqual(3)
})
