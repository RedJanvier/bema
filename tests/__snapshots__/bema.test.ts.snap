// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can define cases in groups 1`] = `
Object {
  "0": Object {
    "caseResultsByMatrix": Array [
      Object {
        "0": Object {
          "report": Object {
            "group": "a",
            "matrix": Array [],
            "name": "name:1",
            "parameters": Object {
              "name": "1",
            },
            "parametersFromMatrix": Object {},
            "parametersFromNotMatrix": Object {
              "name": "1",
            },
            "quickMode": true,
            "stats": "STRIPPED",
          },
        },
        "1": Object {
          "report": Object {
            "group": "a",
            "matrix": Array [],
            "name": "name:2",
            "parameters": Object {
              "name": "2",
            },
            "parametersFromMatrix": Object {},
            "parametersFromNotMatrix": Object {
              "name": "2",
            },
            "quickMode": true,
            "stats": "STRIPPED",
          },
        },
      },
    ],
    "name": "a",
    "time": "STRIPPED",
  },
  "1": Object {
    "caseResultsByMatrix": Array [
      Object {
        "0": Object {
          "report": Object {
            "group": "b",
            "matrix": Array [],
            "name": "name:3",
            "parameters": Object {
              "name": "3",
            },
            "parametersFromMatrix": Object {},
            "parametersFromNotMatrix": Object {
              "name": "3",
            },
            "quickMode": true,
            "stats": "STRIPPED",
          },
        },
        "1": Object {
          "report": Object {
            "group": "b",
            "matrix": Array [],
            "name": "name:4",
            "parameters": Object {
              "name": "4",
            },
            "parametersFromMatrix": Object {},
            "parametersFromNotMatrix": Object {
              "name": "4",
            },
            "quickMode": true,
            "stats": "STRIPPED",
          },
        },
      },
    ],
    "name": "b",
    "time": "STRIPPED",
  },
}
`;
